presto-main/src/main/java/com/facebook/presto/server/PrestoSystemRequirements.java

```diff
@@ -14,14 +14,20 @@
 package com.facebook.presto.server;
 
 import com.google.common.base.StandardSystemProperty;
+import com.sun.management.UnixOperatingSystemMXBean;
 import io.airlift.slice.Slice;
 import io.airlift.slice.Slices;
 import org.joda.time.DateTime;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
 import java.nio.ByteOrder;
 
 final class PrestoSystemRequirements
 {
+
+    private static final int MIN_FILE_DESCRIPTORS = 10_000;
```

Comment:
We were just talking about adding a `-D` system property to override this.  BTW we were just looking at our clusters and average about 5k, and we have seen some peaks near 15k over the past month.